node/pkg/watchers/solana/client.go

```diff
@@ -977,6 +1044,21 @@ func (s *SolanaWatcher) processMessageAccount(logger *zap.Logger, data []byte, a
 		Unreliable:       !reliable,
 	}
 
+	if !reliable && len(observation.Payload) == 0 {
```

Comment:
probably worth a `// SECURITY:` comment on this line. definitely need to not re-observe shim produced messages from the account as this would omit the payload.